src/drivers/power_monitor/ina238/ina238.cpp

```diff
@@ -96,14 +96,19 @@ int INA238::read(uint8_t address, uint16_t &data)
 {
 	// read desired little-endian value via I2C
 	uint16_t received_bytes;
-	const int ret = transfer(&address, 1, (uint8_t *)&received_bytes, sizeof(received_bytes));
+	int ret = PX4_ERROR;
 
-	if (ret == PX4_OK) {
-		data = swap16(received_bytes);
+	for (size_t i = 0; i < 6; i++) {
```

Comment:
Why not use the actual _retries mechanism in I2C?

https://github.com/PX4/PX4-Autopilot/blob/e3b98e6ed23346a13005c69395c6fafb389e9487/src/lib/drivers/device/nuttx/I2C.cpp#L229-L235

Reply:
Updated and pushed.